SKIRT/core/HEALPixSkyInstrument.hpp

```diff
@@ -0,0 +1,170 @@
+/*//////////////////////////////////////////////////////////////////
+////     The SKIRT project -- advanced radiative transfer       ////
+////       © Astronomical Observatory, Ghent University         ////
+///////////////////////////////////////////////////////////////// */
+
+#ifndef HEALPIXSKYINSTRUMENT_HPP
+#define HEALPIXSKYINSTRUMENT_HPP
+
+#include "AllSkyProjection.hpp"
+#include "HomogeneousTransform.hpp"
+#include "Instrument.hpp"
+
+////////////////////////////////////////////////////////////////////
+
+/** The HEALPixSkyInstrument provides an all-sky instrument that does not perform any projection, but instead
+    records fluxes onto a HEALPix tessellation of the sky sphere that guarantees equal surface areas for all
+    pixels.
+
+    The instrument is based on the official HEALPix algorithm by Górski, K. M. et al. (2005)
+    (https://ui.adsabs.harvard.edu/abs/2005ApJ...622..759G/abstract), as available from
+    https://healpix.sourceforge.io, but only uses the relevant part of the algorithm.
+
+    In short, the HEALPix tessellation starts from a base tessellation of the sphere into 12 quadrilaterals.
+    4 quadrilaterals are formed by subdividing the equator into 4 equal segments, so that each quadrilateral
+    has two equatorial segment endpoints as vertices, complemented with two vertices that are obtained by
+    translating the segment midpoint away from the equator over a zenith angle @f$\theta{}=\acos(2/3)@f$ in
+    both directions. The remaining 8 quadrilaterals are formed by connecting the north or south pole of the
+    sphere with 3 vertices of already existing quadrilaterals.
+
+    Within this base tessellation, higher order HEALPix tessellations are constructed by subdividing the
+    12 base quadrilaterals, as if these were simple squares in a flat Euclidian space. To guarantee that all
+    pixels have the same surface area, this subdivision needs to use the same number of sub-pixels along each
+    axis of the base quadrilateral. This number of sub-pixels is referred to as the HEALPix resolution
+    parameter @f$N_{side}@f$. It is common practice within the community to restrict @f$N_{side}=2^k@f$, with
+    @f$k@f$ an integer order, although this is not strictly required. We will use this parametrisation, as it
+    allows for a more efficient pixel location algorithm.
+
+    After subdivision, the HEALPix tessellation will contain $12N_{side}^2$ pixels. These pixels are ordered
+    so that their centres lie on $4N_{side}-1$ rings of equal latitude. On each of these rings, pixel centres
```

Comment:
Please use \f$...\f$ for math in documentation. Just $...$ won't work. I know that @f$...@f$ will work as well, but the convention in SKIRT is to use \f$...\f$. Thanks.